Inc/HALALMock/Models/PinModel/Pin.hpp

```diff
@@ -78,6 +78,65 @@ enum PinState{
 	ON
 };
 
+enum PinType {
+	DigitalOutput,
+	DigitalInput,
+	PWM,
+	ADC
+	// TODO: Add more types
+};
+
+struct PinModel {
+	PinType type; // Always check type before using the union
+
+	union PinData {
+		struct DigitalOutput {
+			// TODO FW-52
+		};
+		struct DigitalInput {
+			// TODO FW-53
+		};
+		struct PWM {
+			// TODO FW-58
+		};
+		struct ADC {
+			// TODO FW-54
+		};
+		// TODO Add more types
+	};
```

Comment:
don't know if you tried to compile this, I think the CMake is not ready to compile with the SIM_ON variable in mind, but the way to do this would be more like:
```c++
union {
    struct {
        // TODO FW-52
    } digital_output;

    struct {
        // TODO FW-53
    } digital_input;

    // ...
} pin_data;
```
The other way around you are declaring a named struct and a named union, but not a member of the union or struct if that makes sense